client/state/data-layer/wpcom/me/notification-settings/index.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { NOTIFICATION_SETTINGS_REQUEST } from 'state/action-types';
+import { receiveNotificationSettings } from 'state/notification-settings/actions';
+import { errorNotice } from 'state/notices/actions';
+
+/**
+ * Dispatches a request to fetch the current user notification settings
+ *
+ * @param   {Function} dispatch Redux dispatcher
+ * @param   {Object}   action   Redux action
+ * @returns {Object}            dispatched http action
+ */
+export const requestNotificationSettings = ( { dispatch }, action ) => dispatch( http( {
+	apiVersion: '1.1',
+	method: 'GET',
+	path: '/me/notifications/settings',
+}, action ) );
+
+/**
+ * Dispatches a notification settings receive action then the request succeeded.
+ *
+ * @param   {Function} dispatch  Redux dispatcher
+ * @param   {Object}   action    Redux action
+ * @param   {Function} next      data-layer-bypassing dispatcher
+ * @param   {Object}   settings  raw notification settings object returned by the endpoint
+ * @returns {Object}             disparched user devices add action
+ */
+export const handleSuccess = ( { dispatch }, action, next, settings ) => dispatch( receiveNotificationSettings( {
+	settings
+} ) );
```

Comment:
there are two things I would love to see here…

 1. have a more descriptive function name describe _what_ the function is intending to do with the response; for example, could be `updateSettings()`

 2. run the `settings` object through some explicit `fromApi()` function to map it between the shape of the data which the API returns and the shape of the data which Calypso actually wants. this can involve renaming data points, combining or extracting lists, converting date formats, etc…